From 06b2fa17e0a3a990f35d8d162aeb47d7ad9f6ea0 Mon Sep 17 00:00:00 2001
From: Patric Stout <truebrain@openttd.org>
Date: Sat, 16 Oct 2021 21:00:02 +0200
Subject: [PATCH 0606/1053] Codechange: "set but not used" warning when
 disabling assert()

DebugCheckSanity() is unused when asserts are disabled. While at it,
use WITH_ASSERT over NDEBUG, as that means we also run this code
during beta/RC.
---
 src/order_cmd.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/order_cmd.cpp b/src/order_cmd.cpp
index 4c94bedc05..f6e7f77a48 100644
--- a/src/order_cmd.cpp
+++ b/src/order_cmd.cpp
@@ -609,6 +609,7 @@ bool OrderList::IsCompleteTimetable() const
 	return true;
 }
 
+#ifdef WITH_ASSERT
 /**
  * Checks for internal consistency of order list. Triggers assertion if something is wrong.
  */
@@ -642,6 +643,7 @@ void OrderList::DebugCheckSanity() const
 			(uint)this->num_orders, (uint)this->num_manual_orders,
 			this->num_vehicles, this->timetable_duration, this->total_duration);
 }
+#endif
 
 /**
  * Checks whether the order goes to a station or not, i.e. whether the
@@ -1782,7 +1784,7 @@ void CheckOrders(const Vehicle *v)
 		/* Do we only have 1 station in our order list? */
 		if (n_st < 2 && message == INVALID_STRING_ID) message = STR_NEWS_VEHICLE_HAS_TOO_FEW_ORDERS;
 
-#ifndef NDEBUG
+#ifdef WITH_ASSERT
 		if (v->orders.list != nullptr) v->orders.list->DebugCheckSanity();
 #endif
 
-- 
2.55.0

